README.md

```diff
@@ -6,27 +6,392 @@
+
+```shell
+poetry run python manage.py migrate
+```
+
+6. Generate test data
+
+```shell
+poetry run python manage.py create_test_data
+```
+
+7. Start the server
+
+```shell
+poetry run manage.py runserver localhost:8000
+```
+
+Backend should now be running at `localhost:8000`.
+
+> Since the backend is not in the root of the project, the source path is not correct for linting out of the box.
+> Fixing this is specific to your IDE, but here are some setups for popular ones:
+>
+> PyCharm: Right click on the `backend` folder and select "Mark Directory as" and then select "Sources Root".
+>
+> VSCode: In your workspace settings (`.vscode/settings.json`), add the following:
+>
+> ```json
+> {
+>     "python.analysis.extraPaths": [
+>         "$(workspaceFolder)/backend"
+>     ]
+> }
+> ```
+
+### Linting
+
+It's recommended to set up [Ruff] linting and formatting support in your editor.
+
+[Ruff]: https://docs.astral.sh/ruff/editors/setup/
+
+### Testing
+
+Tests are run with `pytest`.
+
+Some flags that can save time when running tests:
+
+- To skip slow-running tests: `pytest --skip-slow`
+- To retain test database between runs: `pytest --reuse-db`
+- To skip migration-checks at the start of tests: `pytest --no-migrations`
+- To run tests in parallel: `pytest -n 8 --dist=loadscope` (=8 cores, use `-n auto` to use all available cores)
+
+You can use a `pytest.ini` file to set up flags for local development.
+
+### Updating dependencies
+
+Dependencies are managed by [Poetry]. Normally, they are automatically updated by [dependabot]
+without any manual intervention (given updates don't fail any automated tests).
+
+[Poetry]: https://python-poetry.org/
+[dependabot]: https://github.com/dependabot
+
+However, if you want to update them manually, you can do so by running:
+
+```shell
+poetry update
+```
+
+This will update all dependencies according to the rules defined in `pyproject.toml`.
+To see all outdated dependencies, run:
+
+```shell
+poetry show -o
+```
+
+Note that this will also include any sub-dependencies that are not directly defined in `pyproject.toml`.
+
+### Background tasks
+
+Scheduled & background tasks are run with [Celery].
+
+[Celery]: https://github.com/celery/
+
+When developing locally, you can run these tasks in a Celery worker with `make celery`.
+This uses the filesystem as the message broker.
+You'll need to create queue and processed folders according to the
+`CELERY_QUEUE_FOLDER_OUT`, `CELERY_QUEUE_FOLDER_IN`, `CELERY_PROCESSED_FOLDER`
+environment variables (see `.env.example`).
+
+If you want to run background tasks synchronously without Celery, set the environment variable
+`CELERY_ENABLED` to `False`. Scheduled tasks still need the worker in order to run.
+
+### Authentication
+
+Authentication is handled by [Helsinki Tunnistus] Keycloak using the [django-helusers] library.
+You'll need to get the `TUNNISTAMO_ADMIN_SECRET` from the [Azure Pipelines library]
+or from a colleague and set that in your `.env` file.
```

Comment:
For a more barebones experience, just the django superuser can be used for authentication, but i'm almost sure some features will be broken 🤔 
(This might be useful info if the `TUNNISTAMO_ADMIN_SECRET` is not available, or when just quickly setting up the project)